includes/external-connection-cpt.php

```diff
@@ -539,9 +539,36 @@ function meta_box_external_connection_details( $post ) {
 					<th><?php esc_html_e( 'Can push?', 'distributor' ); ?></th>
 				</thead>
 				<tbody>
-					<?php foreach ( $post_types as $post_type ) : ?>
-						<?php
-						if ( 'dt_subscription' === $post_type->name ) {
+					<?php
+					// Hide the built-in post types except 'post' and 'page'.
+					$hide_from_list = get_post_types(
+						array(
+							'_builtin'     => true,
+							'show_in_rest' => true,
+						)
+					);
+					unset( $hide_from_list['post'], $hide_from_list['page'] );
+
+					// Default is keyed by the post type 'post' => 'post', etc; hence using `array_values`.
+					$hide_from_list = array_values( $hide_from_list );
+
+					// Hide 'dt_subscription' post type.
+					$hide_from_list[] = 'dt_subscription';
+
+					/**
+					 * Filter to update the list of post types that should be hidden from the "Post types permissions" list.
+					 *
+					 * @since x.x.x
+					 * @hook dt_hide_post_types_from_permission_list
+					 *
+					 * @param {array} The list of post types..
```

Comment:
```suggestion
					 * @param {array} The list of hidden post types.
```

Nitpick. I'll merge to avoid delaying approval.